### ISSUE TITLE
Shutdown not working

### ISSUE THREAD

**Issue by MrGreg1** [1000001101000/acp-commander#10]
My device is a TeraStation TS5600D running the latest firmware 4.18-0.08. I am trying to figure out how to shutdown the device from command line. I discovered acp-commander. I cannot get the device to shutdown. Windows Firewall was disabled for testing. Here is what I have tried.

java -jar acp_commander.jar -t x.x.x.x -shutdown    [WARN] Using a random connection ID without authentication!
java -jar acp_commander.jar -t x.x.x.x -pw password -shutdown   [WARN] Using a random connection ID without authentication!
java -jar acp_commander.jar -t x.x.x.x -o -pw password -shutdown   Resetting root password... Password changed. Starting telnetd...Sucess!

The last one cleared my root password which I did not want to do. I should of been more careful. I would like to reset the password to factory. However I do not know what it was. Do you know what it is?

I have also tried the GUI version 1.56 for testing. I am able to authenticate but the shutdown option does not succeed.

What am I doing wrong? Is this even possible given my model and firmware version?

**Comment by 1000001101000** [comment]
Good to hear from you.

To be honest, I haven’t looked much at the shutdown command. It wouldn’t surprise me to learn that it didn’t work with all models/firmware versions.

I would expect sending a generic shutdown shell command would work, something like:
`java -jar acp_commander.jar -t x.x.x.x -pw password -c “shutdown now”`

I don’t know what the default root password is. I think the best thing to do would be to set it to something secure/random. After that you can kill off telnetd or reboot to close telnet access.

**Comment by MrGreg1** [comment]
Hi.

Thanks for your speedy response! I have made some progress. 

I called Buffalo and inquired about the factory root password. They claim to not know what it is. I used the GUI to set the root password. It was just easier. So that is taken care of.

I was able to use the GUI version to get the shutdown to work. The problem was I was putting -shutdown in the command box. That is wrong. It should be just shutdown.

Once I had that working, I was able to get it to work using your command line version. The -c switch did the trick. However it is not "shutdown now" It is just shutdown. Here is the command that works.

java -jar acp_commander.jar -t x.x.x.x -pw password -c shutdown

However there is still a firewall problem. The GUI version does not have any problem passing the Windows Firewall But the command line version does. It only works with the firewall disabled. If enabled, I get

A SocketTimeoutException was thrown which usually indicates bad firewall settings

This is interesting because the firewall has 2 inbound rules for Java. One for TCP and the other for UDP. Both allow all ports local and remote.

Do you have any idea why the GUI version passes the firewall and the command line does not?

Thanks for your support...

**Comment by 1000001101000** [comment]
The `admin` password and the `root` password are different things. The default admin password is typically "password" and can be changed via the admin portal. The root password isn't normally known/used and gets set to blank by `-o` to make getting a root shell over telnet simple. Having a blank root password is a bit of a security risk, i'd recommend connecting again and setting it to something secure with `passwd` to be on the safe side. 

There are a few different types of communication used by acp_commander and Buffalo's tools like NasNavigator and TSUpdater even though they use basically the same protocol for most operations. This includes udp broadcast both incoming and outgoing, tcp and udp. I believe it's all on that same port.

**Comment by Neustradamus** [comment]
@MrGreg1: What is the situation with your TS5600D?

**Comment by MrGreg1** [comment]
Hi Neustradamus, This has been resolved. The problem was Oracle Java. It was not allowing acp-commander to pass the windows firewall. I used an online utility to convert the jar file to exe. That passed the firewall. Then a windows update broke that as well. I switched Oracle Java to Eclipse Temurin JRE open source. Now both the original jar and the converted exe pass the windows firewall. I had to add a UDP and TCP rule for JRE of course. The switch also fixed a problem with my Ubiquiti Waps. The management app uses java as well. Thanks for reaching out!
